File: backend/app/models/product.py
from sqlalchemy import Column, Integer, String, Text, DECIMAL, TIMESTAMP, ForeignKey
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.db.database import Base


class Product(Base):
    __tablename__ = "products"

    product_id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    category_id = Column(Integer, ForeignKey("categories.category_id"))
    product_name = Column(String(150), nullable=False)
    brand = Column(String(100))
    price = Column(DECIMAL(12, 2), nullable=False)
    stock = Column(Integer, default=0)
    image_url = Column(String(255))
    description = Column(Text)
    created_at = Column(TIMESTAMP, server_default=func.current_timestamp())

    # Relationships
    category = relationship("Category", back_populates="products")
    order_details = relationship("OrderDetail", back_populates="product")
    reviews = relationship("Review", back_populates="product")

File: frontend/src/types/product.ts
export interface ProductSummaryDto {
  productId: number;
  categoryId: number;
  categoryName: string;
  productName: string;
  brand?: string | null;
  price: number;
  stock: number;
  imageUrl?: string | null;
  description?: string | null;
  createdAt: string;
  isInStock: boolean;
}

export interface ProductDetailDto extends ProductSummaryDto {
  updatedAt?: string | null;
}

export interface ProductCategoryDto {
  categoryId: number;
  categoryName: string;
}

export interface ProductCardModel {
  id: number;
  name: string;
  price: number;
  image: string;
  category: string;
  brand?: string | null;
  stock: number;
}

export interface ProductDetailModel extends ProductCardModel {
  brand?: string | null;
  description?: string | null;
  createdAt: string;
  updatedAt?: string | null;
}

export interface ProductCategory {
  id: number;
  name: string;
}